client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -65,20 +67,31 @@ tie.factory('PythonPrereqCheckService', [
     var detectAndGetWrongLanguageType = function(code) {
       var codeLines = getNonStringLines(code);
 
+      var rawCodeArray = code.split('\n');
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
+        var firstErrorCharIndex = code.search(regexp);
+        var errorLine = null;
 
-        if (error.allowMultiline) {
-          if (code.search(regexp) !== -1) {
-            return error.errorName;
+        if (firstErrorCharIndex !== -1) {
+          // Match found, Subtract each line.length until error line pops.
+          for (var j = 0; firstErrorCharIndex > 0; j++) {
+            var sub = rawCodeArray[j].length;
+            firstErrorCharIndex -= sub;
           }
-        } else {
-          for (var j = 0; j < codeLines.length; j++) {
-            if (codeLines[j].search(regexp) !== -1) {
-              return error.errorName;
+          errorLine = firstErrorCharIndex;
```

Comment:
Why are we setting errorLine to firstErrorCharIndex? Not sure I understand the logic here...

Reply:
Corrected. It should be the index variable, not firstErrorCharIndex.
Done.